homeassistant/components/eq3btsmart/climate.py

```diff
@@ -181,18 +474,53 @@ def preset_modes(self):
 
         Requires SUPPORT_PRESET_MODE.
         """
-        return list(HA_TO_EQ_PRESET)
+        if self._sensor_control:
+            return PRESET_NONE
 
-    def set_preset_mode(self, preset_mode):
+        return list(HA_TO_EQ_PRESET.keys())
+
+    async def async_set_preset_mode(self, preset_mode: str):
         """Set new preset mode."""
+        if self._sensor_control:
+            return
+
         if preset_mode == PRESET_NONE:
-            self.set_hvac_mode(HVAC_MODE_HEAT)
-        self._thermostat.mode = HA_TO_EQ_PRESET[preset_mode]
+            await self.async_set_hvac_mode(HVAC_MODE_HEAT)
+
+        self.set_thermostat_mode(HA_TO_EQ_PRESET[preset_mode])
 
     def update(self):
         """Update the data from the thermostat."""
 
+        _LOGGER.debug("%s - Updating thermostat", self._name)
         try:
             self._thermostat.update()
         except BTLEException as ex:
-            _LOGGER.warning("Updating the state failed: %s", ex)
+            _LOGGER.warning("%s - Updating the state failed: %s", self._name, ex)
+
+    def set_thermostat_mode(self, mode):
+        """Set the thermostat mode."""
+
+        _LOGGER.debug(
+            "%s - Setting thermostat mode to %s", self._name, EQ_TO_STRING[mode]
+        )
+        try:
+            self._thermostat.mode = mode
+        except BTLEException as ex:
+            _LOGGER.warning("%s - Setting the state failed: %s", self._name, ex)
+
+    def turn_off(self):
+        """Turn off thermostat."""
+
+        _LOGGER.info("%s - Turning off thermostat", self._name)
+        self.set_thermostat_mode(eq3.Mode.Closed)
+
+    def turn_on(self):
+        """Turn on thermostat."""
+
+        if self._hvac_mode == HVAC_MODE_OFF:
+            _LOGGER.error("%s - Invalid turn on request", self._name)
+            return
+
+        _LOGGER.info("%s - Turning on thermostat", self._name)
```

Comment:
```suggestion
        _LOGGER.debug("%s - Turning on thermostat", self._name)
```
I think the consensus is not to log regular happenings at info level.

Reply:
Just a note, there were also other places where this was done incorrectly, I just didn't want to pollute the review too much :-)